data_transfer/services/ucam.py

```diff
@@ -98,12 +98,38 @@ def record_by_wear_period(
     """
     If data was created on a certain period then it belongs to an individual patient.
     """
-    device_wear_periods = device_history(device_id)
+    records = device_history(device_id)
+
+    return __record_in_wear_period(records, start_wear, end_wear)
+
+
+def record_by_wear_period_in_list(
+    devices: List[Device], start_wear: datetime, end_wear: datetime
+) -> Optional[Payload]:
+    """
+    If data was created on a certain period then it belongs to an individual patient.
+    """
+    return __record_in_wear_period(devices, start_wear, end_wear)
+
+
+def __record_in_wear_period(
+    records: List, start_wear: datetime, end_wear: datetime
+) -> Optional[Payload]:
+    """
+    Shared method given the logic is the same above
+    """
+    start_wear = normalise_day(start_wear)
+    end_wear = normalise_day(end_wear)
+
+    for record in records:
+        # While we could compare records to the second, this caused some issues, e.g.,
+        # when a record was created on the day for testing but not checked out until afterwards.
+        drm_start_wear = normalise_day(record.start_wear)
+        drm_end_wear = normalise_day(record.end_wear)
+
+        within_start_period = drm_start_wear <= start_wear <= drm_end_wear
```

Comment:
Although this method is currently solely used for Dreem, I would argue to keep the naming device agnostic. I.e. `drm_start_wear` could be `record_start_wear`